src/dotnet/Db/DbSetExt.cs

```diff
@@ -26,4 +26,199 @@ public static class DbSetExt
             throw new ArgumentOutOfRangeException(nameof(key), "Key must be greater than zero.");
         return set.FindAsync(DbKey.Compose(key), cancellationToken);
     }
+
+    public static async Task Lock<TEntity, TKey>(
+        this DbSet<TEntity> set,
+        TKey key,
+        CancellationToken cancellationToken)
+        where TEntity : class
+    {
+        var context = set.GetDbContext();
+        var keyHash = (long)(key?.GetHashCode() ?? 239);
+        var lockKey = (keyHash << 32) ^ HashCode.Combine(typeof(TEntity).GetHashCode());
+        var timeout = context.Database.GetCommandTimeout();
+        // TODO(AK): find a way to resolve DBSettings instance
```

Comment:
You can make DBSettings.LockTimeout property static so far or move this to constants.